Common/Data/Custom/Tiingo/TiingoDailyData.cs

```diff
@@ -0,0 +1,189 @@
+        /// </summary>
+        [JsonProperty("adjVolume")]
+        public long AdjustedVolume { get; set; }
+
+        /// <summary>
+        /// The dividend paid out on "date" (note that "date" will be the "exDate" for the dividend)
+        /// </summary>
+        [JsonProperty("divCash")]
+        public decimal Dividend { get; set; }
+
+        /// <summary>
+        /// A factor used when a company splits or reverse splits. On days where there is ONLY a split (no dividend payment),
+        /// you can calculate the adjusted close as follows: adjClose = "Previous Close"/splitFactor
+        /// </summary>
+        [JsonProperty("splitFactor")]
+        public decimal SplitFactor { get; set; }
+
+        /// <summary>
+        /// Return the URL string source of the file. This will be converted to a stream
+        /// </summary>
+        /// <param name="config">Configuration object</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>String URL of source file.</returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            DateTime startDate;
+            if (!_startDates.TryGetValue(config.Symbol.Value, out startDate))
+            {
+                startDate = date;
+                _startDates.TryAdd(config.Symbol.Value, startDate);
+            }
+
+            var source = $"https://api.tiingo.com/tiingo/daily/{config.Symbol.Value}/prices?startDate={startDate:yyyy-MM-dd}&token={_authCode}";
+            return new SubscriptionDataSource(source, SubscriptionTransportMedium.RemoteFile, FileFormat.Collection);
+        }
+
+        /// <summary>
+        ///     Reader converts each line of the data source into BaseData objects. Each data type creates its own factory method,
+        ///     and returns a new instance of the object
+        ///     each time it is called. The returned object is assumed to be time stamped in the config.ExchangeTimeZone.
+        /// </summary>
+        /// <param name="config">Subscription data config setup object</param>
+        /// <param name="content">Content of the source document</param>
+        /// <param name="date">Date of the requested data</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        ///     Instance of the T:BaseData object generated by this line of the CSV
+        /// </returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string content, DateTime date, bool isLiveMode)
+        {
+            var list = JsonConvert.DeserializeObject<List<TiingoDailyData>>(content);
+
+            foreach (var item in list)
+            {
+                item.Symbol = config.Symbol;
+                item.Time = item.Date;
```

Comment:
cool, looks like we're properly handling start/end times of the bar here -- I'm assuming that `item.Date` means the item contains daily data for that date